### PR TITLE
Help guides link

### PR DESCRIPTION
**Pull request by gtardif**
**- What I did**
* Add a link to docker help guides (at docs.docker.com) for general help messages 
* gate this with an env var to revert to no link if users want to avoid this additional message

link actually depends on https://github.com/docker/docker.github.io/pull/11812

**- How I did it**
Added section at the bottom of go template, and a function to check if it should display or not

**- How to verify it**
`docker help` will display it, `docker --help` also, but not `docker run --help` 

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/cobra.go
Comment (thaJeztah) [inline cli/cobra.go:210]: Not sure if we need this; I think overall adding content to the `help` output should be ok

File: cli/cobra.go
Comment (thaJeztah) [inline cli/cobra.go:380]: Wondering (discussed that option briefly with @chris-crone) if instead of hard-coding, we should use annotations similar to; https://github.com/docker/cli/blob/v20.10.0-rc1/cli/cobra.go#L195-L198

That way we could just print the information if its set on a specific (or "any") command

File: cli/cobra.go
Comment (gtardif) [reply cli/cobra.go:210]: cc @nebuk89 can we drop this requirement ? (I agree this doesn't seem too invasive)

File: cli/cobra.go
Comment (chris-crone) [reply cli/cobra.go:380]: +1 to this but we can also do it as a follow up if we add links for specific helps

Comment (gtardif) [reply cli/cobra.go]: Right, added the additional help message in root annotation, so we can add specific messages to any command as we like in the future. (also renamed methods for something more homogeneous

File: cli/cobra.go
Comment (gtardif) [reply cli/cobra.go]: removed this env var check with other changes (comment below), we can add it back later if needed

Comment (nebuk89) [reply cli/cobra.go]: let's leave it out for now... I agree that it's good practice but this kind of feels like something help has just been missing rather than something very pushy :)

File: cli/cobra.go
Comment (chris-crone) [inline cli/cobra.go:52]: super nit: @thaJeztah do we need the trailing slash

File: cli/cobra.go
Comment (chris-crone) [inline cli/cobra.go:222]: @nebuk89 can you confirm what is the right color? I think the spec said "light cyan" which might be "bright cyan" [here](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors), i.e.: 96

**Review by chris-crone — APPROVED** [review]
Two questions but LGTM

File: cli/cobra.go
Comment (nebuk89) [reply cli/cobra.go:222]: Ah yes I meant bright sorry!

Comment (gtardif) [reply cli/cobra.go]: Updated

**Comment by gtardif** [comment]
@thaJeztah should be OK now I guess ; still dependent on merging the redirect in docs.docker.com, though (and @chris-crone comment on the trailing slash).

File: cli/cobra.go
Comment (gtardif) [reply cli/cobra.go]: updated

**Comment by chris-crone** [comment]
@thaJeztah Got the OK from ben on this. Can we remove the trailing slash?

**Comment by nebuk89** [comment]
LGTM ship it!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by tiborvass — APPROVED** [review]

**Comment by gtardif** [comment]
https://github.com/docker/docker.github.io/pull/11812 has been merged, anything blocking this one or should it be merged ?  (@chris-crone @thaJeztah)

**Pull request merged**
